taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -328,24 +374,37 @@ const ListTranslations = ({ nodeObject, setNodeObject }) => {
                         if (!language) language = '';
                         setNewLanguageCode(ISO6391.getCode(language));
                         const isValidLanguage = ISO6391.validate(ISO6391.getCode(language))
+                        const isAlreadyShown = shownTranslationLanguages.includes(ISO6391.getCode(language))
                         const isDuplicateLanguage = renderedTranslations.some(el => (el.languageCode === ISO6391.getCode(language))) || 
                                                 nodeObject.main_language === ISO6391.getCode(language)
-                        if (isValidLanguage && !isDuplicateLanguage) {setisValidLanguageCode(true)}
-                        else {setisValidLanguageCode(false)}
+                        if (isValidLanguage  && !isDuplicateLanguage) {setLanguageAction("add")}
+                        else if (isValidLanguage && !isAlreadyShown && isDuplicateLanguage) {setLanguageAction("show")}
+                        else {setLanguageAction("invalid")}
                     }}
-                    renderInput={(params) => 
+                    renderOption={(props, option) => {
+                        const languageCode = ISO6391.getCode(option)
+                        const isAlreadyShown = addedLanguageCodes.includes(languageCode)
+                        // set color of text to green if the language is already shown
+                        return (
+                            <span {...props} style={{ color: isAlreadyShown ? 'green' : 'black' }}>
+                                {isAlreadyShown ? <b>{option}</b> : option}
+                            </span>
+                        )
+                    }}
+                    renderInput={(params) =>
                         <TextField
-                            error={!isValidLanguageCode} {...params} label="Languages" 
+                            error={languageAction === "invalid"} {...params} label="Languages"
                         />}
                 />
                 </DialogContent>
                 <DialogActions>
                 <Button onClick={handleClose}>Cancel</Button>
-                <Button 
-                    disabled={!isValidLanguageCode}
-                    onClick={() => {handleAddTranslation(newLanguageCode)}}>
-                        Add
-                </Button>
+                { languageAction !== "invalid" && 
+                    <Button
+                        onClick={() => {languageAction === "add" ? handleAddTranslation(newLanguageCode) : handleToggleTranslationVisibility(newLanguageCode)}}>
+                            {languageAction === "add" ? "Add" : "Show" }
+                    </Button>
+                }
```

Comment:
If languageAction is invalid, I think we could add a warning message instead of the button, so that user understand better the dynamic.

```suggestion
                { languageAction !== "invalid" && 
                    <Button
                        onClick={() => {languageAction === "add" ? handleAddTranslation(newLanguageCode) : handleToggleTranslationVisibility(newLanguageCode)}}>
                            {languageAction === "add" ? "Add" : "Show" }
                    </Button> 
                    ||
                    <Button disabled>Already selected</Button>
                }
```

Reply:
![image](https://user-images.githubusercontent.com/41837037/211043687-7d071fec-e3b6-4491-9f87-2ba006449170.png)
The invalid state itself can have multiple reasons 😅 , like not selecting any language.
Should I just put `Select hidden language` here

Reply:
it's ok, like this :-)